clients/go/pkg/audit/interceptor.go

```diff
@@ -278,6 +278,40 @@ func (i *Interceptor) StreamInterceptor(srv interface{}, ss grpc.ServerStream, i
 	}
 	logReq.Payload.AuthenticationInfo = &capi.AuthenticationInfo{PrincipalEmail: principal}
 
+	// If JWT Validator hasn't been added, we don't handle justification JWTs, and just log
+	// without the justification.
+	if i.jwtValidator != nil {
+		// JWT Validator added, look for justification info
+		md, ok := grpcmetadata.FromIncomingContext(ctx)
+		if !ok {
+			return i.handleReturnStream(ctx, ss, handler, status.Errorf(codes.FailedPrecondition, "unable to retreive grpc metadata."))
+		}
+		vals := md.Get("justification_token")
+		if len(vals) == 0 {
+			return i.handleReturnStream(ctx, ss, handler, status.Errorf(codes.Internal,
+				"no justification token found."))
+		}
+		jwtRaw := vals[0]
+		tok, err := i.jwtValidator.ValidateJWT(jwtRaw)
+		if err != nil {
+			return i.handleReturnStream(ctx, ss, handler, status.Errorf(codes.Internal,
+				"audit interceptor failed converting parsing or validating justification token: %v", err))
+		}
+		buf, err := json.Marshal(*tok)
+		if err != nil {
+			return i.handleReturnStream(ctx, ss, handler, status.Errorf(codes.Internal,
+				"couldn't convert token to json: %v", err))
+		}
+		// Note: We don't set metadata before here, so we can directly set it.
+		// If we need to put other data in metadata, this should be modified to not
+		// overwrite it.
+		logReq.Payload.Metadata = &structpb.Struct{
+			Fields: map[string]*structpb.Value{
+				"justification_token": structpb.NewStringValue(string(buf)),
```

Comment:
Setting as string is not query friendly. I think we should convert it a `*structpb.Value`, see example: https://github.com/abcxyz/lumberjack/blob/main/clients/go/pkg/audit/runtime.go#L251

Reply:
Lets do that as a follow-up, as the java code does this as well right now. I think it makes sense to upgrade all if we decide to change the format.

Reply:
sgtm. Please create an issue.

Reply:
https://github.com/abcxyz/lumberjack/issues/265